lib/json-schema/validator.rb

```diff
@@ -64,42 +70,14 @@ def initialize(schema_data, data, opts={})
         # Don't clear the cache during metaschema validation!
         self.class.validate!(metaschema, @base_schema.schema, {:clear_cache => false})
       end
-
-      # If the :fragment option is set, try and validate against the fragment
-      if opts[:fragment]
-        @base_schema = schema_from_fragment(@base_schema, opts[:fragment])
-      end
     end
 
     def schema_from_fragment(base_schema, fragment)
       schema_uri = base_schema.uri
-      fragments = fragment.split("/")
```

Comment:
another fix I just noticed: this should be `split("/", -1)`; as is empty string keys would be mishandled. `#/foo//` should evaluate to fragments `['foo', '', '']` and in json `{"foo": {"": {"": 7}}}` should evaluate to `7`.